src/main/kotlin/org/rust/ide/typing/paste/RsConvertJsonToStructCopyPasteProcessor.kt

```diff
@@ -196,22 +196,113 @@ private fun createAndInsertStruct(
     return inserted
 }
 
-fun generateStruct(
+private fun Char.isSeparator(): Boolean = this == '_' || this == '-'
+
+// fooBarBaz
+private fun String.isCamelCase(): Boolean =
+    isNotEmpty()
+        && first().isLowerCase()
+        && !contains("__")
+        && !contains("--")
```

Comment:
Tbh I'm not sure if these conditions are needed, I just copied them from similar logic in the naming lint.